### DISCUSSION TITLE
Limitations on PBI Report Server and AS 2019

### DISCUSSION THREAD

**Discussion by burtbinia91** [TabularEditor/TabularEditor3#978] [🙋♀️TE3 Q&A]
Hello,

*EDIT* I have just found the solution to my problem with the Debug feature. If I delete the automatic FormatString generated by the Calculation Groups in my model everything works as anticipated. This limitation is ofc. also noted within the documentation of TE 3, I just didn't make enough sense of it before posting.  So for anyone else with the same problem just remove the automatic FormatString before execution.

![image](https://github.com/TabularEditor/TabularEditor3/assets/115256235/e6f9312c-4c01-44b0-ba09-9ea77098877b)

*Original Post*
I have been trying to evaluate Tabular Editor 3 for our company and was really looking forward to the debug features in TE 3. I have been using TE2 for some time to work with Calculation Groups but was missing some of the ease of life features which come with TE3.

So far i have tried to debug some simple queries via the DAX Querys generated by the PBI performance analyzer. Sadly i was not able to debug a single one as debugging was never availabe for any of the querys regardless of the complexitiy. I created the querys on a fresh page without any filters or Calculation Groups on them. Never the less there **are** Calculation Groups in our Model with Format Expressions which ofc. duplicate any Measure in the Model. Could this be a reason?

Then I tried the Pivot Grid feature and again I ran into an error whenever I tried to use any column of the model as either a row or a column stating: " Error occured durin processing server request (The given Key xx was not present in the dictionary)"

So im wondering now if there are any limitation regarding our development environment which is using PBI Report Server on the January 2023 Update aswell as Analysis Service 2019 with a live connection.

Thank you for any insight or hint in the right direction if I am doing something worng.

BR,
Burt

**Comment by otykier** [comment]
Hi Burt,
Thanks for updating us with your findings. We will investigate if there's something we can do to make debugging work, even when dynamic format strings are present in the query.
